File: src/test/java/com/meghneelgore/geometry/sorting/TestKNearest.java
package com.meghneelgore.geometry.sorting;

import com.meghneelgore.geometry.Point;
import org.junit.Assert;
import org.junit.Test;

import java.util.ArrayList;
import java.util.List;

public class TestKNearest {

    @Test
    public void testFindKNearest() {
        Point p1 = new Point(1.0, 1.0);
        Point p2 = new Point(2.0, 2.0);
        Point p3 = new Point(3.0, 3.0);
        Point p4 = new Point(-1.5, -1.5);

        ArrayList<Point> input = new ArrayList<>();
        input.add(p1);
        input.add(p2);
        input.add(p4);
        input.add(p3);

        KNearest kNearest = new KNearest();
        List<Point> list = kNearest.findKNearest(input, 2);

        Assert.assertEquals("Wrong number of points in list", 2, list.size());
        Assert.assertEquals("Wrong point in list", new Point(-1.5, -1.5), list.get(0));
        Assert.assertEquals("Wrong point in list", new Point(1.0, 1.0), list.get(1));
    }
}

File: src/test/java/com/meghneelgore/geometry/sorting/TestKNearest.kt
package com.meghneelgore.geometry.sorting

import com.meghneelgore.geometry.primitives.Point
import org.junit.Assert
import org.junit.Test

import java.util.ArrayList

class TestKNearest {

    @Test
    fun testFindKNearest() {
        val p1 = Point(1.0, 1.0)
        val p2 = Point(2.0, 2.0)
        val p3 = Point(3.0, 3.0)
        val p4 = Point(-1.5, -1.5)

        val input = ArrayList<Point>()
        input.add(p1)
        input.add(p2)
        input.add(p4)
        input.add(p3)

        val kNearest = KNearest()
        val list = kNearest.findKNearest(input, 2)

        Assert.assertEquals("Wrong number of points in list", 2, list.size.toLong())
        Assert.assertEquals("Wrong point in list", Point(-1.5, -1.5), list[0])
        Assert.assertEquals("Wrong point in list", Point(1.0, 1.0), list[1])
    }
}
